src/pybind/mgr/cephadm/tests/test_cephadm.py

```diff
@@ -475,6 +476,37 @@ def test_extra_container_args(self, _run_cephadm, cephadm_module: CephadmOrchest
                     image='',
                 )
 
+    @mock.patch("cephadm.serve.CephadmServe._run_cephadm")
+    def test_custom_config(self, _run_cephadm, cephadm_module: CephadmOrchestrator):
+        _run_cephadm.side_effect = async_side_effect(('{}', '', 0))
+        test_cert = ['-----BEGIN PRIVATE KEY-----',
+                     'YSBhbGlxdXlhbSBlcmF0LCBzZWQgZGlhbSB2b2x1cHR1YS4gQXQgdmVybyBlb3Mg',
+                     'ZXQgYWNjdXNhbSBldCBqdXN0byBkdW8=',
+                     '-----END PRIVATE KEY-----',
+                     '-----BEGIN CERTIFICATE-----',
+                     'YSBhbGlxdXlhbSBlcmF0LCBzZWQgZGlhbSB2b2x1cHR1YS4gQXQgdmVybyBlb3Mg',
+                     'ZXQgYWNjdXNhbSBldCBqdXN0byBkdW8=',
+                     '-----END CERTIFICATE-----']
+        configs = [
+            CustomConfig(content=['testting', 'testing', 'is', 'this', 'on'],
```

Comment:
oops, I guess since lists of strings are also json serializable and it's not actually trying to write anything out in this test they still passed with a list of strings. Fixed the tests and also added a validation function to the CustomConfig class to have it error if gets something that isn't a string.